app/scripts/lib/xhr.js

```diff
@@ -67,6 +67,7 @@ define([
         // make sure to set the dataType for Firefox <21. See issue #1930
         dataType: 'json',
         type: options.type,
+        timeout: options.timeout || 5000,
```

Comment:
Let's not set a default timeout here just yet, this could have unintended side effects when interfacing with other services. We know Basket is a problem now, let's address just that first.

You can transform this to pass `timeout` if it's defined in `options` doing something like so:

```js
var request = {...};
if ('timeout' in options) {
  request.timeout = options.timeout;
}
```